src/convertToObject.js

```diff
@@ -1,12 +1,21 @@
 'use strict';
 
-/**
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+
+  const arr = sourceString.trim().split(';');
+
+  for (let i = 0; i <= arr.length; i++) {
+    const elem = arr[i];
+
+    if (elem && elem.trim() && elem.split(':')) {
+      const [key, value] = elem.split(':');
```

Comment:
Such a check is very unproductive why would you do a split twice? It is better to do it once, save it to a variable and use it in both cases

Reply:
good idea, will do. Thank you.